learn/architecture.md

```diff
@@ -53,5 +53,67 @@ When the contract is reposted, some of the faulty node's collateral pays for the
 
 ![architect](/learn/architecture.png)
 
+### Marketplace architecture ###
+
+The marketplace consists of a smart contract that is deployed on-chain, and the sales and purchasing modules that are part of the node software. The sales module is responsible for posting storage requests to the smart contract. The purchasing module is its counterpart that storage providers use to determine which storage requests they are interested in.
```

Comment:
It's the other way around: purchasing is used by clients to post storage requests, and sales is used by SPs to fill slots and provide storage.